### ISSUE TITLE
ERROR: No matching distribution found for extractcode-7z>=16.5.210525; extra == "full" on Mac M1

### ISSUE THREAD

**Issue by swastkk** [nexB/scancode-toolkit#3748]
### Description

Getting this error while running the startup command
```
./configure
```
Error
``` 
ERROR: Could not find a version that satisfies the requirement extractcode-7z>=16.5.210525; extra == "full" (from extractcode[full]) (from versions: none)
ERROR: No matching distribution found for extractcode-7z>=16.5.210525; extra == "full"
```

### How To Reproduce

> Tell us how to reproduce the issue. 

### System configuration

> For bug reports, it really helps us to know:

* What OS are you running on? (Windows/MacOS/Linux) 
macOS 14.4.1 23E224 arm64

* What version of scancode-toolkit was used to generate the scan file?  
scancode-toolkit-32.1.0

* What installation method was used to install/run scancode? (pip/source download/other) 
Installed via cloning scancode-toolkit with --depth=1

**Comment by swastkk** [comment]
We need to document this in the scancode-toolkit documentation, just like https://github.com/nexB/scancode.io/pull/894/commits/cf7441a7db5d68b26e31e05240063cb7dbd06f5c so that mac M1 users don't face much problem.
The steps mentioned by @OmkarPh & @keshav-space  in community discussion 

https://matrix.to/#/!PutoaZVYVzoQjfaxVC:gitter.im/$59jSfXANKInWABANASlId8QaxZ9g26haxjVOevtrfYk?via=gitter.im&via=matrix.org&via=fsfe.org
is easy and smooth and should be included in the documentation.

**Comment by tsteenbe** [comment]
@swastkk  Running into this same issue on a M2 Pro - tried following instructions from Matrix link but no luck. Was able to get ScanCode installed via `pip` see https://github.com/nexB/scancode-toolkit/issues/3626#issuecomment-2084527200

**Comment by heliocastro** [comment]
The actual full package cause a pip loop that tries to install recursively all scancode installs. Is similar on what happens in Linux Arm64, so is not exclusive to MacOS, but Arm environments itself.
I tried originally to debug the issue, but enter in a huge rabbit hole that not got near the real issue.
I narrowed down to extractcode and deps on it, but stopped there and unfortunately never found time to continue ( sorry @pombredanne )
